### ISSUE TITLE
InternetStatus changing frequently

### ISSUE THREAD

**Issue by abhinavsinghring** [OutdatedGuy/internet_connection_checker_plus#51]
Thank you for this package. 
Everything is working as expected but returning InternetStatus.disconnected even if there is an internet connection.
I've encountered this during my video player page where videos are loading perfectly from the cloud and when I try to skip the video it shows me the No connection overlay (which I've added so that the user can't use the app when there is no internet). But this is causing frequent overlay popup. 

I think this is due to the disconnection between the user and website this package is checking the connection for whenever there is slow internet.

Is there any way to fix this or I've to remove this overlay?

network_controller.dart
```
import 'dart:async';
import 'dart:developer';

import 'package:get/get.dart';
import 'package:internet_connection_checker_plus/internet_connection_checker_plus.dart';
import 'package:paperbook/Screens/Home/Controller/home_controller.dart';
import 'package:paperbook/Screens/NoNetwork/network_screen.dart';
import 'package:paperbook/Services/StreakController/streak_controller.dart';


class NetworkController extends GetxController {
  static NetworkController get instance => Get.find();

  RxBool isConnectedToInternet = false.obs;
  StreamSubscription? _internetConnectionStreamSubscription;
  Timer? _checkNetworkScreenTimer;
  RxBool isNoNetworkScreenShown = false.obs; // Flag to track if NoNetworkScreen is shown

  @override
  void onInit()async{
    super.onInit();
    isConnectedToInternet.value = await InternetConnection().hasInternetAccess;
    log("NetStatus: ${isConnectedToInternet.value}");
    _internetConnectionStreamSubscription = 
      InternetConnection().onStatusChange.listen((event) {
        log("$event");
        switch (event) {
          case InternetStatus.connected:
            isConnectedToInternet.value=true;
            _removeNoNetworkScreen();
            break;
          case InternetStatus.disconnected:
            isConnectedToInternet.value=false;
            _showNoNetworkScreen();
            break;
          default:
           isConnectedToInternet.value=false;
        }
      });
  }

  @override
  void onClose(){
    _internetConnectionStreamSubscription?.cancel();
    _checkNetworkScreenTimer?.cancel();
    super.onClose();
  }

  void _showNoNetworkScreen() {
    if (Get.currentRoute != '/NoNetworkScreen' && isNoNetworkScreenShown.value == false) {
      Get.to(
        ()=> const NoNetworkScreen(),
        routeName: '/NoNetworkScreen',
        opaque: false,
        fullscreenDialog: true,
        preventDuplicates: true,
        duration: const Duration(milliseconds: 500),
        transition: Transition.downToUp
      );
      isNoNetworkScreenShown.value = true; // Set the flag to true when the screen is shown
      log("ScreenStatus: ${isNoNetworkScreenShown.value}");
    } else {
      null;
    }
  }

  void _removeNoNetworkScreen() {
    _checkNetworkScreenTimer?.cancel();
    _checkNetworkScreenTimer = Timer.periodic(const Duration(seconds: 2), (timer) {
      if (Get.currentRoute == '/NoNetworkScreen' && !Get.isSnackbarOpen && isConnectedToInternet.value) {
        Get.back();
        isNoNetworkScreenShown.value = false; // Reset the flag when the screen is removed
        log("ScreenStatus: ${isNoNetworkScreenShown.value}");
        timer.cancel(); // Stop the timer once the condition is met
      }
    });
  }

  void removeNoNetworkScreenRetry() {
    if (isConnectedToInternet.value) {
      _checkNetworkScreenTimer?.cancel();
      _checkNetworkScreenTimer = Timer.periodic(const Duration(seconds: 2), (timer) {
        if (Get.currentRoute == '/NoNetworkScreen' && isConnectedToInternet.value) {
          Get.back();
          isNoNetworkScreenShown.value = false; // Reset the flag when the screen is removed
          log("ScreenStatus: ${isNoNetworkScreenShown.value}");
          timer.cancel(); // Stop the timer once the condition is met
        }
      });
    }
  }
}
```
network_screen.dart
```
import 'package:flutter/material.dart';
import 'package:gap/gap.dart';
import 'package:get/get.dart';
import 'package:paperbook/Constants/size_constant.dart';
import 'package:paperbook/Services/Connectivity/network_controller.dart';
import 'package:paperbook/Widgets/Text/custom_text.dart';

import '../../Constants/color_constant.dart';
import '../../Constants/size_config.dart';
import '../Downloads/download_screen.dart';

class NoNetworkScreen extends StatelessWidget {
  const NoNetworkScreen({super.key});

  @override
  Widget build(BuildContext context) {
    SizeConfig().init(context);
    final NetworkController networkController = Get.put(NetworkController());
    return Scaffold(
      body: PopScope(
        canPop: false,
        child: Padding(
          padding: EdgeInsets.all(mainpadding/1.2),
          child: Column(
            mainAxisAlignment: MainAxisAlignment.start,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              Gap(mainpadding*5),
              Image.asset("assets/icons/disconnected.png", height:mainpadding*12, width:mainpadding*12),
              Gap(mainpadding*2),
              TextWidget(text: "Whoops! You are Offline", size: headingSize28*1.25, center: true, weight: FontWeight.w700),
              Gap(mainpadding/1.5),
              TextWidget(
                text: "No Internet Connection Found\nor internet is slow.\nCheck your connection", 
                size: subHeadingSize+1, 
                center: true, 
                weight: FontWeight.w500,
                color: Colors.grey,
              ),
              Gap(mainpadding),
              OutlinedButton(
                onPressed: (){
                  networkController.removeNoNetworkScreenRetry();
                }, 
                style: OutlinedButton.styleFrom(
                  minimumSize: Size(mainpadding*7, SizeConfig.safeBlockVertical!*5.5),
                  side: BorderSide(
                    color: Theme.of(context).colorScheme.tertiary
                  )
                ),
                child: TextWidget(
                  text: "Refresh",
                  size: subHeadingSize,
                  weight: FontWeight.w500,
                  color: Theme.of(context).colorScheme.tertiary
                ),
              ),
              Container(),
            ],
          ),
        )
      ),
      bottomNavigationBar: Padding(
        padding: EdgeInsets.only(left: mainpadding/1.2, right: mainpadding/1.2, bottom: mainpadding),
        child: ElevatedButton(
          style: ElevatedButton.styleFrom(
            minimumSize: Size(double.infinity, SizeConfig.safeBlockVertical! * 6)
          ),
          onPressed: ()async{
            await Get.to(
              () => const DownloadsPage(),
              routeName: '/DownloadsPage',
              transition: Transition.cupertino,
              duration: const Duration(milliseconds: 500),
              popGesture: true,
              gestureWidth: (context) {
                double width =
                    MediaQuery.of(context).size.width;
                return width;
              },
            );
          },
          child: TextWidget(
            text: "Go to Downloads",
            size: subHeadingSize,
            color: white,
            weight: FontWeight.w500
          ),
        ),
      ),
    );
  }
}
```

**Comment by OutdatedGuy** [comment]
Hi @abhinavsinghring, can you run your on web and send screenshot of network tab? This'll help us see what is going wrong.

**Comment by abhinavsinghring** [comment]
Thank you for the response.
But I have changed my way of handling things when there is disconnected status. Now I'm just showing a stripe in appBar bottom when there is changes and user can continue using app except server related things.
Also I have moved to different place now and here internet connection is good so I can't create that problem naturally but I think that happens because of the slow internet.
